### ISSUE TITLE
Votes from banned users should be removed

### ISSUE THREAD

**Issue by kazk** [Codewars/codewars.com#1807]
Votes for satisfaction and rank from banned spammers/cheaters should be removed when the user is banned so the negative effects don't persist.

https://github.com/Codewars/codewars.com/issues/1378#issuecomment-488534812

**Comment by Blind4Basics** [comment]
same apply to rank suggestions (sounds like the jerk just learnt the button was accessible, today...)

**Comment by Bubbler-4** [comment]
I recently got offensive votes (satisfaction rating and estimated rank) from fsafsffs on [this kata](https://www.codewars.com/kata/domino-tiling-m-x-n-board-with-holes/javascript). I assume the account is banned (0 honor). I reverted the offending tag, but can't revert the estimated rank because there's an 8kyu vote. (Before that, my own estimated rank was 2kyu.)

If no action is possible right now, I could just abandon it and create a copy of it, since *all* of the existing solutions are cheats (most of them even didn't submit final).

**Comment by kazk** [comment]
@Bubbler-4 I've fixed yours manually for now.

**Comment by Blind4Basics** [comment]
Hi kazk,

You resolved my comment above, but I'm not sure about what it's supposed to mean ("ok, I got the info", or "this has been handled").

The thing is, I still can see the problem on my [FMC2](https://www.codewars.com/kata/full-metal-chemist-number-2-parse-me-dot-dot-dot) kata ;) : 8kyu rank and 1 of the not satisfied vote are from our day-to-day hater (the other down vote is from a cheater too...). That's actually the only thing that is forbidding the approval of it (was approvable even with the first down vote).

Note that the stats of [Bubbler's kata](https://www.codewars.com/kata/domino-tiling-m-x-n-board-with-holes) are a bit weird after your intervention: "100% of 0" at the top, but still showing up 1 down vote at the bottom of the page. So I don't know what's the best course of action here. Waiting for a "safer fix"? fix manually? Your call. I just wanted to let you know.

Cheers.

**Comment by kazk** [comment]
@Blind4Basics I updated the issue body based on your comment (added "for satisfaction and rank") and resolved it.

For FMC2, is it the same user? I was able to manually fix Bubbler's because I knew which user to look for.

> "100% of 0" at the top, but still showing up 1 down vote at the bottom of the page.

Thanks for letting me know. Looks like some callback wasn't called because I updated the value in the database without using the method on the model. Banned user is not allowed to vote or change their vote so I need to figure out a workaround for this.

I'll try to do add this feature during this week for a proper fix.
